## Data Analyst Assessment 
### Social Data

In [1]:
import pandas as pd

In [2]:
# read the dataset into a panda Dataframe
data = pd.read_excel("~/Downloads/social_data.xlsx")

In [4]:
data.head(10)

Published Date Account Account Type          Campaign Name  \
0 2023-03-31 19:55:43.192    CSGO      TWITTER                   N/A    
1 2023-03-31 19:49:35.243    CSGO      TWITTER                   N/A    
2 2023-03-31 19:49:33.535    CSGO      TWITTER                   N/A    
3 2023-03-31 19:49:31.583    CSGO      TWITTER                   N/A    
4 2023-03-31 19:43:42.590    CSGO      TWITTER  Community Engagement    
5 2023-03-31 19:42:11.331    CSGO      TWITTER                   N/A    
6 2023-03-31 19:40:57.383    CSGO      TWITTER  Community Engagement    
7 2023-03-31 19:23:57.349    CSGO      TWITTER  Community Engagement    
8 2023-03-31 19:15:14.431    CSGO      TWITTER                   N/A    
9 2023-03-31 18:55:51.855    CSGO      TWITTER  Community Engagement    

   Total Impressions  Total Engagements Media Type  
0                  0                  0       Text  
1                  0                  0       Text  
2                  0                  0       Text  
3                  0                  0       Text  
4               9517               1215      Video  
5                  0                  0      Photo  
6               1882                 83      Photo  
7              17892               3076      Video  
8                  0                  0      Video  
9              13874               1447      Video

#### 1. What is the typical engagement rate we can expect? What's the likelihood that we can achieve a 15% engagement rate?

In [5]:
# Filter the dataset with only the 'Total Impressions' being non-zero
filtered_data = data[data['Total Impressions'] > 0]


In [6]:
filtered_data

Published Date   Account Account Type          Campaign Name  \
4    2023-03-31 19:43:42.590      CSGO      TWITTER  Community Engagement    
6    2023-03-31 19:40:57.383      CSGO      TWITTER  Community Engagement    
7    2023-03-31 19:23:57.349      CSGO      TWITTER  Community Engagement    
9    2023-03-31 18:55:51.855      CSGO      TWITTER  Community Engagement    
10   2023-03-31 18:36:01.000  General        FBPAGE  Community Engagement    
...                      ...       ...          ...                    ...   
3463 2023-01-04 13:17:28.473  General       TWITTER                   N/A    
3464 2023-01-04 13:00:02.187  General       TWITTER  Community Engagement    
3465 2023-01-04 13:00:01.000  General        FBPAGE  Community Engagement    
3470 2023-01-03 13:34:33.887  General       TWITTER                   N/A    
3477 2023-01-03 10:34:03.438      CSGO      TWITTER                   N/A    

      Total Impressions  Total Engagements Media Type  
4                  9517               1215      Video  
6                  1882                 83      Photo  
7                 17892               3076      Video  
9                 13874               1447      Video  
10                37692               8044      Photo  
...                 ...                ...        ...  
3463                393                 18       Text  
3464             136660               2371      Photo  
3465              29051                811      Photo  
3470               7704                153       Text  
3477               2116                 42       Text  

[2735 rows x 7 columns]

In [7]:
# calculate the engagement rate
filtered_data['Engagement Rate'] = (filtered_data['Total Engagements'] / filtered_data['Total Impressions']) * 100

/var/folders/fs/4bsx3pp90m99tqxgzq6j1s6m0000gn/T/ipykernel_98979/1527339301.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_data['Engagement Rate'] = (filtered_data['Total Engagements'] / filtered_data['Total Impressions']) * 100


In [8]:
# calculate the typical engagement rate
typical_engagement_rate = filtered_data['Engagement Rate'].mean()

In [10]:
# Print the result
print("Typical Engagement rate is {:.2f}%".format(typical_engagement_rate))

Typical Engagement rate is 40.49%


In [11]:
# Assess the likelihood of achieving a 15% engagement rate
likelihood = len(filtered_data[filtered_data['Engagement Rate'] >= 15]) / len(filtered_data) * 100

In [12]:
# Print the likelihood
print("Likelihood of achieving a 15% engagement rate: {:.2f}%".format(likelihood))

Likelihood of achieving a 15% engagement rate: 8.26%


#### Answer: So the typical engagement rate we can expect is 40.49% and the likelihood that we can achieve a 15% engagement rate is 8.26%

#### 2. Does day of the week and time of posting affect engagement rates?

In [14]:
filtered_data.dtypes

Published Date       datetime64[ns]
Account                      object
Account Type                 object
Campaign Name                object
Total Impressions             int64
Total Engagements             int64
Media Type                   object
Engagement Rate             float64
dtype: object

In [15]:
# Extract the day of week and the time of posting
filtered_data['Day of Week'] = filtered_data['Published Date'].dt.dayofweek
filtered_data['Time of Posting'] = filtered_data['Published Date'].dt.time

/var/folders/fs/4bsx3pp90m99tqxgzq6j1s6m0000gn/T/ipykernel_98979/673600769.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_data['Day of Week'] = filtered_data['Published Date'].dt.dayofweek
/var/folders/fs/4bsx3pp90m99tqxgzq6j1s6m0000gn/T/ipykernel_98979/673600769.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_data['Time of Posting'] = filtered_data['Published Date'].dt.time


In [16]:
filtered_data

Published Date   Account Account Type          Campaign Name  \
4    2023-03-31 19:43:42.590      CSGO      TWITTER  Community Engagement    
6    2023-03-31 19:40:57.383      CSGO      TWITTER  Community Engagement    
7    2023-03-31 19:23:57.349      CSGO      TWITTER  Community Engagement    
9    2023-03-31 18:55:51.855      CSGO      TWITTER  Community Engagement    
10   2023-03-31 18:36:01.000  General        FBPAGE  Community Engagement    
...                      ...       ...          ...                    ...   
3463 2023-01-04 13:17:28.473  General       TWITTER                   N/A    
3464 2023-01-04 13:00:02.187  General       TWITTER  Community Engagement    
3465 2023-01-04 13:00:01.000  General        FBPAGE  Community Engagement    
3470 2023-01-03 13:34:33.887  General       TWITTER                   N/A    
3477 2023-01-03 10:34:03.438      CSGO      TWITTER                   N/A    

      Total Impressions  Total Engagements Media Type  Engagement Rate  \
4                  9517               1215      Video        12.766628   
6                  1882                 83      Photo         4.410202   
7                 17892               3076      Video        17.192041   
9                 13874               1447      Video        10.429581   
10                37692               8044      Photo        21.341399   
...                 ...                ...        ...              ...   
3463                393                 18       Text         4.580153   
3464             136660               2371      Photo         1.734963   
3465              29051                811      Photo         2.791642   
3470               7704                153       Text         1.985981   
3477               2116                 42       Text         1.984877   

      Day of Week  Time of Posting  
4               4  19:43:42.590000  
6               4  19:40:57.383000  
7               4  19:23:57.349000  
9               4  18:55:51.855000  
10              4         18:36:01  
...           ...              ...  
3463            2  13:17:28.473000  
3464            2  13:00:02.187000  
3465            2         13:00:01  
3470            1  13:34:33.887000  
3477            1  10:34:03.438000  

[2735 rows x 10 columns]

In [17]:
# Group the data by day of the week and time of posting, and calculate the average engagement rate
grouped_data = filtered_data.groupby(['Day of Week', 'Time of Posting'])['Engagement Rate'].mean().reset_index()

In [22]:
grouped_data_sorted = grouped_data.sort_values(by='Engagement Rate', ascending=False)

In [25]:
grouped_data

Day of Week  Time of Posting  Engagement Rate
0               0         06:00:08         4.661169
1               0         06:00:09         3.125000
2               0         06:06:04         1.303781
3               0         06:06:12         1.013047
4               0  06:47:27.379000         2.857143
...           ...              ...              ...
2676            6  23:09:08.803000         2.487562
2677            6  23:09:36.760000         1.913876
2678            6  23:13:48.866000         1.706485
2679            6  23:14:11.149000         6.611570
2680            6  23:16:47.935000         1.754386

[2681 rows x 3 columns]

In [28]:
grouped_data_sorted.head(20)

Day of Week  Time of Posting  Engagement Rate
1820            4         12:37:06     94000.000000
1226            3         11:55:44      1258.666667
1041            3         06:02:34       880.000000
1046            3         06:18:27       100.000000
1382            3  15:00:00.377000        60.000000
624             2         07:25:18        58.858403
1335            3         14:25:33        57.889590
1040            3         06:02:28        57.500000
185             1         05:18:48        56.480028
2653            6         17:48:41        55.765629
2669            6         20:48:33        51.560997
1345            3         14:36:46        45.396165
555             1         16:54:36        44.993383
1781            4         11:41:25        44.713813
902             2         14:00:01        43.092220
1052            3         07:00:00        41.077418
1456            3         15:50:18        40.040880
1039            3         06:02:22        40.000000
1127            3         09:22:33        38.448840
437             1         13:46:26        38.082478

In [30]:
# Count the number of data points with engagement rate > 15
total_high_engagement = len(grouped_data[grouped_data['Engagement Rate'] > 15])

# Print the result
print(total_high_engagement)


206


In [27]:
# Count occurrences of high engagement rates by day of the week
high_engagement_count = grouped_data[grouped_data['Engagement Rate'] > 15]['Day of Week'].value_counts()

# Print the result
print(high_engagement_count)


3    53
1    41
4    39
6    31
2    20
0    11
5    11
Name: Day of Week, dtype: int64


In [31]:
# Count occurrences of high engagement rates by time of posting
high_engagement_time_count = grouped_data[grouped_data['Engagement Rate'] > 15]['Time of Posting'].value_counts()

# Print the result
print(high_engagement_time_count)


09:00:01    2
17:42:21    2
09:00:57    1
11:10:53    1
08:59:07    1
           ..
06:02:22    1
06:02:28    1
06:02:34    1
06:16:55    1
20:48:33    1
Name: Time of Posting, Length: 204, dtype: int64


In [32]:
# Extract the hour from the 'Time of Posting' column
grouped_data['Hour of Posting'] = grouped_data['Time of Posting'].apply(lambda x: x.hour)

# Count occurrences of high engagement rates by hour of posting
high_engagement_hour_count = grouped_data[grouped_data['Engagement Rate'] > 15]['Hour of Posting'].value_counts()

# Print the result
print(high_engagement_hour_count)


9     31
16    23
14    21
17    18
13    17
12    16
8     15
11    13
15    13
7     10
10     8
6      7
18     6
5      3
21     2
4      1
19     1
20     1
Name: Hour of Posting, dtype: int64


#### Answer: 
#### So from the above analyzing, we can see that the daya of week will affect the engagement rates whereas the exact time of posting will not normally have impact to the engagement rate. Based on the analysis above, we can see that day 3 which corresponds to Wednesday has the highest count among the rest of the week. And the exact time doesn't have a lot overlap, but when we only think about the hour of the time, 9 o'clock appears to have the highest count that has the engagment rate greater than 15. That suggest that Wednesday and 9 o'clock tend to have higher occurrence of high engagement rates compared to other days in the dataset. But these are specific to the given dataset, with other dataset it might varies. 

#### 3. How our game title doing in terms of social performance? Is there a specific game we should focus more on or less?

In [34]:
# group the dataset by the game title and also calculate the total engagements
game_performance = data.groupby("Account")["Total Engagements"].sum().sort_values(ascending=False)


In [35]:
# print the results
print("Total Engagements by Game Title:")
print(game_performance)

Total Engagements by Game Title:
Account
General             3076479
DOTA2                123990
CSGO                  92336
Content Creators       1387
Valorant                931
General                 849
Name: Total Engagements, dtype: int64


In [36]:
# return the game title with the highest total engagements
top_game = game_performance.idxmax()
print("Top Game Title:", top_game)

Top Game Title: General 


In [37]:
# return the game title with the lowest total engagements
bottom_game = game_performance.idxmin()
print("Bottom Game Title:", bottom_game)

Top Game Title: General


#### Answer:
#### Based on the analysis I have, there are two apperance of "General" may results the data inconsistency or unspecify of the game and all in category General. Based on the dataset, I think if we can specify specific games instead of using general, we might have different results and can give a more valuable suggestion about which game to focus more or less since there do have valueable games in category of General needed to focus more and also some games results in low engagements that can be focus less. But without more detailed data, it is hard to find out futher suggestion. But besides General, we can see DOTA2 is the second highest engagement rate and Content Creators and Valorant are having lower social performance, which can suggested that DOTA2 need more focus and relatively can spend less focus on Valorant and followed by Content Creators.

#### 4. What media type performs the best?

In [38]:
media_engagements = data.groupby("Media Type")["Total Engagements"].sum().sort_values(ascending=False)
print(media_engagements)


Media Type
Photo       2167739
Video        859451
Text         246312
Mixed         13664
Carousel       6541
Link           2257
Album             8
Name: Total Engagements, dtype: int64


#### Answer:
#### Similarly, we can come with Photo is the media type that performs the best.

#### 5. What is our best performing campaign?

In [40]:
best_campaign = data.groupby("Campaign Name")["Total Engagements"].sum().sort_values(ascending=False).head(1).index[0]
print(best_campaign)

Community Engagement 


#### Answer:
#### With the above code, we can find out that 'Community Engagement' is our best performing campaign

#### 6. Define out a posting strategy for our social channels based on your discoveries.

According to the analysis above, I believe the first strategy should put more of an emphasis on community engagement because this campaign has the highest level of engagement. Give more time, money, and effort to producing content and events that promote social interaction and neighborhood involvement. Second, since images and moving pictures are considered to be the most captivating media types, we should place special emphasis on the visual content. Prioritizing the creation of visually appealing and compelling content can greatly aid in grabbing the audience's attention. But since there are also many people who believe these are interesting media types, the followed by high orrcurity of other types of text, mixed, and carousal also suggested to make the posting divisive. Next, the various game preferences suggested that more effort should be put into promoting hit games like General, DOTA2, and CSGO. Last but not least, posting at the right time is a crucial strategy. According to the data, Wednesday between the hours of 9 and 16 appears to be a good time to post in order to increase visibility and interaction.

#### 7. What suggestions would you give to the social media team if they want to expand their presence (e.g. if our CSGO youtube channel is doing well should we expand to TikTok)?

If the social media team want to expand their presence, I think there are few steps needed to consider. The first step is to conduct market research because it's crucial to understand whether or not our target market is active and using TikTok. Interests, demographics, and engagement patterns are just a few of the audience characteristics that need to be investigated. Second, because every platform has a different strategy for drawing users, it is crucial to consider how we want to express our game on TikTok. The success of a YouTube channel does not guarantee success on TikTok. Since TikTok is known for its brief videos and ease of accessibility, our YouTube postings may need to be modified in order to conform to the TikTok trends.  Due to the success and popularity of video content, there are ways to reuse or create original content from the YouTube channel for TikTok. The most important step in extending to a new platform is this one because it involves content creation, community management, monitoring, keeping up with trends on the new platform, and finding new ways to post. The final step after considering all of these suggestions is to keep an eye on TikTok's performance. Follow the metrics, such as social performance, engagements, and audience feedback, on a regular basis. That might helps with expanding the presence. 